File: YukariLauncher/src/main/java/net/kdt/pojavlaunch/JAssets.java
package net.kdt.pojavlaunch;

import androidx.annotation.Keep;

import com.google.gson.annotations.SerializedName;

import java.util.Map;

/**
 * マインクラフトのアセットインデックスを保持するクラス。
 */
@Keep
public class JAssets {
    /* 古いバージョンのMCで使用され、ファイルが .minecraft/resources 以下に配置されていた場合 */
    @SerializedName("map_to_resources") public boolean mapToResources;
    /** アセットオブジェクトのマップ */
    public Map<String, JAssetInfo> objects;

    /* legacy.json（〜1.6.X）アセットファイルで使用され、.minecraft/assetsフォルダのルートのパスに使用されます */
    public boolean virtual;
}

File: YukariLauncher/src/main/java/com/arata/yukarilauncher/value/JAssets.kt
package com.arata.yukarilauncher.value

import androidx.annotation.Keep
import com.google.gson.annotations.SerializedName
import com.arata.yukarilauncher.value.JAssetInfo

/**
 * Minecraftのアセットインデックス情報を保持するクラス。
 * アセットのマッピング設定やオブジェクト一覧を格納します。
 */
@Keep
class JAssets {
    /** アセットをリソースにマッピングするかどうかのフラグ。 */
    @SerializedName("map_to_resources") var mapToResources: Boolean = false
    /** アセットオブジェクトのマップ。キーはアセットパス、値はアセット情報。 */
    var objects: MutableMap<String, JAssetInfo>? = null
    /** 仮想アセットかどうかのフラグ。 */
    var virtual: Boolean = false
}